synapse/cortex.py

```diff
@@ -834,24 +878,29 @@ def getCoreMods(self):
             ret.append((modname, mod.conf))
         return ret
 
+    @s_coro.genrhelp
     async def eval(self, text, opts=None, user=None):
         '''
         Evaluate a storm query and yield Nodes only.
         '''
+        await self.boss.promote('storm', user=user, info={'query': text})
         async with await self.snap(user=user) as snap:
-            async for node in snap.eval(text, opts=opts):
+            async for node in snap.eval(text, opts=opts, user=user):
                 yield node
 
+    @s_coro.genrhelp
     async def storm(self, text, opts=None, user=None):
         '''
-        Evaluate a storm query and yield result messages.
+        Evaluate a storm query and yield (node, path) tuples.
         Yields:
             (Node, Path) tuples
         '''
+        await self.boss.promote('storm', user=user, info={'query': text})
```

Comment:
Do we wnat to include opts here?

Reply:
No, unfortunately, opts might contain an "idens" key with a zillion things in it :)